meveo-admin/ejbs/src/main/java/org/meveo/service/base/BusinessService.java

```diff
@@ -255,15 +269,19 @@ public void addFilesToModule(P entity, MeveoModule module) throws IOException, B
     	String businessEntityDtoSerialize = JacksonUtil.toString(businessEntityDto);
     	
     	File gitDirectory = GitHelper.getRepositoryDir(currentUser, module.getGitRepository().getCode());
-    	String path = businessEntityDto.getClass().getAnnotation(ModuleItem.class).path() + "/" + entity.getCode();
+    	String path = entity.getClass().getAnnotation(ModuleItem.class).path() + "/" + entity.getCode();
     	
     	File newDir = new File (gitDirectory, path);
-    	
     	newDir.mkdirs();
     	
+    	File newJsonFile = new File(gitDirectory, path+"/"+entity.getCode()+".json");
+    	newJsonFile.createNewFile();
+
     	byte[] strToBytes = businessEntityDtoSerialize.getBytes();
-    	
-    	Files.write(newDir.toPath(), strToBytes);
+
+    	Files.write(newJsonFile.toPath(), strToBytes);
+
+		gitClient.commitFiles(meveoRepository, Collections.singletonList(newDir), "Add JSON file for entity " + entity.getCode());
```

Comment:
You should not use the meveoRepository, but rather the repository of the concerned module